sdk/typescript/connect.ts

```diff
@@ -28,7 +28,7 @@ export async function connection(
   fct: () => Promise<void>,
   cfg: ConnectOpts = {},
 ) {
-  // Wrap connection into the otpl context
+  // Wrap connection into the otlp context
   await opentelemetry.context.with(getContext(), async () => {
     await defaultContext.connection(cfg)
```

Comment:
We need to make sure all the telemetry stuff is only executed if there's any `OTEL_` env variables set. Something like this:

https://github.com/dagger/dagger/blob/0aba66c0ddc170667b8c87f030e6a6eb38b8c1a2/sdk/go/telemetry/init.go#L24-L31

`dagger run` sets `TRACEPARENT`, which should still be propagated to request headers if it exists, but it doesn't set any other `OTEL_` env vars so we should skip configuring the exporter and everything in that case.